### ISSUE TITLE
Getting wrong weather forecast

### ISSUE THREAD

**Issue by p3k** [regolith-linux/regolith-i3xrocks-config#95]
Don’t know if this is the right place but the i3xrocks weather plugin gives me this URL for the weather forecast:

<https://wttr.in/Vienna,Vienna>

While it looks unsuspiciously on first sight, it actually is the Vienna in Clark County, South Dakota, 57271, United States of America – not the one I live in, the capital of Austria.

The string is composed via this line in  `/usr/share/i3xrocks/weather`:

```sh
IP_LOCATION="$(curl -s https://ipapi.co/city/),$(curl -s https://ipapi.co/region/)"
```

[Igor Chubin](https://twitter.com/p3k/status/1338485033351458816), the author of wttr.in gave some alternatives for naming a place, and so I tried it with `Austria,Austria,AT` which gives the [correct forecast](https://wttr.in/Vienna,Vienna,Austria)!

I.e. the line would work for me like this:

```sh
IP_LOCATION="$(curl -s https://ipapi.co/city/),$(curl -s https://ipapi.co/region/),$(curl -s https://ipapi.co/country)"
```

Now is the script handled by Regolith or would I have to turn to somewhere else to ask for this fix?

**Comment by gbaims** [comment]
I get wrong weather too, but for different reason. As those are related, maybe it's better to write about it in this issue.

I live in São Paulo. Tilde makes wttr unable to find the city and show a random location:

> We were unable to find your location
> so we have brought you to Oymyakon,
> one of the coldest permanently inhabited locales on the planet.

URL generated by the script: https://wttr.in/S%E3o%20Paulo%2CSao%20Paulo
Correct URL: https://wttr.in/Sao%20Paulo%2CSao%20Paulo

**Comment by moritzheiber** [comment]
This is likely fixed by #88. Let us know if it isn't by re-opening the issue.

**Comment by gbaims** [comment]
Hey Moritz. Unfortunetly this pull request won't fix the tilde / accent issue.

Before: https://wttr.in/S%E3o%20Paulo%2CSao%20Paulo
After: https://wttr.in/S%E3o%20Paulo%2CSao%20Paulo%2C01000%2CBR

Should be: https://wttr.in/Sao%20Paulo%2CSao%20Paulo%2C01000%2CBR

I think everyone that has this kind of character in location get wrong weather forecast.

**Comment by moritzheiber** [comment]
Drat, I'll take a look at writing a bunch of tests for the blocklet today and then re-writing the blocklet to pass them

**Comment by moritzheiber** [comment]
Folks, I've added #96. Please take a look and see whether it "fixes" your issues. Thanks!

**Comment by gbaims** [comment]
Yes! New script works here in São Paulo.
